### PR TITLE
[BugFix] Disable row mode partial update when sort key columns are incomplete

### PR DESCRIPTION
**Pull request by sevev**
We don't support row mode partial update if sort key columns are incomplete so far except for the pure delete task. So we will check the `__op` column is `UPSERT` or not when we find we are doing a partial update with incomplete sort key columns.  But when we lost `__op` column, the check will fail, this pr fixes the bug.  And now the check logic is following:
1. If we do a column mode partial update task, sort key columns are not necessary.
2. If we do a row mode partial update task which all sort key columns are provided, we could support upsert and delete.
3. If we do a row mode partial update which sort key columns are incomplete, and `__op` column exists, we will check the operation when we write data. If we find `UPSERT`, the task will fail.
4. If we do a row mode partial update which sort key columns are incomplete, and `__op` column not exists, we will reject the task because when the 'op' column is missing, it defaults to an upsert task.

BTW, This bug has no real impact right now.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 1 / 12 (08.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/delta_writer.cpp|1|12|08.33%|[245, 328, 329, 330, 331, 332, 333, 334, 335, 339, 340]|

File: be/src/storage/delta_writer.cpp [hunk 1]
```diff
@@ -323,16 +324,20 @@ void DeltaWriter::_set_state(State state, const Status& st) {
...
-                LOG(WARNING) << "table with sort key do not support partial update";
-                return Status::NotSupported("table with sort key do not support partial update");
+    if (_tablet->updates() != nullptr && _partial_schema_with_sort_key) {
+        if (_opt.slots != nullptr && _opt.slots->back()->col_name() == "__op") {
+            size_t op_column_id = chunk.num_columns() - 1;
+            const auto& op_column = chunk.get_column_by_index(op_column_id);
+            auto* ops = reinterpret_cast<const uint8_t*>(op_column->raw_data());
+            for (size_t i = 0; i < chunk.num_rows(); i++) {
+                if (ops[i] == TOpType::UPSERT) {
+                    LOG(WARNING) << "table with sort key do not support partial update";
```
Comment (TszKitLo40) [inline be/src/storage/delta_writer.cpp:334]: L334 and L335 are duplicated with L339 and L340.

Comment (chaoyli) [inline be/src/storage/delta_writer.cpp:334]: Add the tablet_id in the log

File: be/src/storage/delta_writer.cpp [hunk 2]
```diff
@@ -323,16 +324,20 @@ void DeltaWriter::_set_state(State state, const Status& st) {
...
+                    return Status::NotSupported("table with sort key do not support partial update");
+                }
             }
+        } else {
+            LOG(WARNING) << "table with sort key do not support partial update";
```
Comment (chaoyli) [inline be/src/storage/delta_writer.cpp:339]: Add the tablet_id in the log

File: be/src/storage/delta_writer.cpp (hunk 1 above)
Comment (sevev) [reply be/src/storage/delta_writer.cpp]: > Add the tablet_id in the log

Done.

File: be/src/storage/delta_writer.cpp (hunk 2 above)
Comment (sevev) [reply be/src/storage/delta_writer.cpp]: > Add the tablet_id in the log

Done

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 1 / 16 (06.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/storage/delta_writer.cpp|1|16|06.25%|[245, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 341, 342, 343, 344]|

**Pull request closed without merging**